File: Solutions OOP/problem 16/Option 01/BoxTest.java
/**
 * @author T.cruiz.
 * @since  Nov 2022.
 */ 

class Box{
    double length;
    double width;
    double height;

    public Box(double h, double w, double l) {
        height = h;
        width = w;
        length = l;
    }

    public void calculateVolume() {
        System.out.println("The volume of a box is : " + (height*width*length));
    }
}

public class BoxTest {
    public static void main(String[] args) {
        Box pcBox = new Box(5, 4, 7);
        pcBox.calculateVolume();
    }
}

File: problem 16/Option 01/BoxTest.kt
class Box(val length: Double, val width: Double, val height: Double) {
    fun calculateVolume(): Double {
        return length * width * height
    }
}


fun main() {
    val box1 = Box(10.0, 20.0, 30.0)
    val volume1 = box1.calculateVolume()
    println("Volume of box1: $volume1")

    val box2 = Box(5.0, 10.0, 15.0)
    val volume2 = box2.calculateVolume()
    println("Volume of box2: $volume2")

    val box3 = Box(3.0, 6.0, 9.0)
    val volume3 = box3.calculateVolume()
    println("Volume of box3: $volume3")
}
